### PR TITLE
Instana tracer implementation

### LINKED ISSUE #2876: Instana tracing backend
Since you have implemented the OpenTracing API now, I wanted to ask if you would be open to accepting a PR adding support for the [Instana golang sensor](https://github.com/instana/golang-sensor) as a tracing backend.

### PR DESCRIPTION
**Pull request by notsureifkevin**
### What does this PR do?

This change adds the Instana tracer as an opentracing compatible backend

### Motivation

I use both traefik and Instana, and this seemed like a worthwhile change to upstream

Fixes  #2876

### More

- [ ] Added/updated tests (note: there are no tests implemented for specific tracing backends)
- [x] Added/updated documentation

### Additional Notes
Traefik is awesome

### CODE REVIEW

**Comment by ldez** [comment]
Hello, thanks for you contribution :+1: 

We accept only bug fixes on 1.7 https://github.com/containous/traefik#release-cycle

Your PR seems interesting, could you rebase on master?
Thanks.

File: docs/configuration/tracing.md [hunk 1]
```diff
@@ -181,3 +181,41 @@ Traefik supports three tracing backends: Jaeger, Zipkin and DataDog.
...
+
+```toml
+# Tracing definition
+[tracing]
+  # Backend name used to send tracing data
+  #
+  # Default: "jaeger"
+  #
+  backend = "instana"
+  # Service name used in DataDog backend
```
Comment (mmatur) [inline docs/configuration/tracing.md]:
```suggestion
  # Service name used in Instana backend
```

Comment (notsureifkevin) [reply docs/configuration/tracing.md]: @mmatur thank you for the review. i've resolved the suggestion and pushed a new revision.

**Comment by remotezygote** [comment]
How do you use this?

I tried the config given as an example in the docs in this PR (btw, why does none of that show up in the actual docs?), but I get the following in logs:

**Comment by dtomcej** [comment]
@remotezygote This PR was merged into master, not 1.7. It appears in the 2.0 release.

For assistance with configuration, reach out on our slack: traefik.slack.com!

Thanks!

**Pull request merged**